src/Aspire.Dashboard/Authentication/FrontendCompositeAuthenticationHandler.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using Aspire.Dashboard.Authentication.Connection;
+using Aspire.Dashboard.Configuration;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Dashboard.Authentication;
+
+public sealed class FrontendCompositeAuthenticationHandler(
+    IOptionsMonitor<DashboardOptions> dashboardOptions,
+    IOptionsMonitor<FrontendCompositeAuthenticationHandlerOptions> options,
+    ILoggerFactory logger,
+    UrlEncoder encoder)
+        : AuthenticationHandler<FrontendCompositeAuthenticationHandlerOptions>(options, logger, encoder)
+{
+    protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        var result = await Context.AuthenticateAsync(ConnectionTypeAuthenticationDefaults.AuthenticationSchemeFrontend).ConfigureAwait(false);
+        if (result.Failure is not null)
+        {
+            return AuthenticateResult.Fail(
+                result.Failure,
+                new AuthenticationProperties(new Dictionary<string, string?>(),
+                new Dictionary<string, object?> { [AspirePolicyEvaluator.SuppressChallengeKey] = true }));
```

Comment:
Indentation's bit off here. Named arguments would be helpful for me too.

```suggestion
            return AuthenticateResult.Fail(
                result.Failure,
                new AuthenticationProperties(
                    items: new Dictionary<string, string?>(),
                    parameters: new Dictionary<string, object?> { [AspirePolicyEvaluator.SuppressChallengeKey] = true }));
```

Also the `items` parameter is nullable, so can we avoid allocating an empty dictionary here?

Reply:
AuthenticationProperties allocates a dictionary if you pass in null anyway.